### DISCUSSION TITLE
do we need in instructions the OP_DEC parameter? I doubt it.

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#374] [Q&A]
for example: `instr_f("dlt", "Decimal Less than op1=(op2<op3)", OP_REG, OP_REG, OP_DEC);`
The OP_DEC operand indicates that the 3. parameter is a directly entered decimal number, which can't be anything else as a floating-point string. So I need to be convinced that I am wrong. For the moment, I am sticking with OP_FLOAT instead of OP_DEC.

**Comment by adesutherland** [comment]
I think in the meantime that is fine. Remember, however, that your instruction implementation is converting from double (the assembler created the double from the literal, noting your issue with its lack of precision in its conversion!) to Decimal. But it will work from the get-go. 

When we can, we will change the OP_FLOAT to OP_DEC and change the double to either the DecPacked struct or to Decimal (TBC). This means that the assembler will need to use the Decimal library, and in the interpreter instructions we will need to replace the conversion from Double.

**Reply by Peter-Jacob** [reply]
The only advantage I can see is if your dec value has more than 15 digits. But wouldn't it be better to keep the decimal as a string and convert it to dec format when needed using the dec library functions?

**Reply by adesutherland** [reply]
I guess that is a third option, OP_DEC would store a string representation of the number (but validated by the assembler).

**Reply by adesutherland** [reply]
Maybe the stored string value would be normalised by the assembler. Anyway, double will work in the meantime
